NuKeeper.Gitea/GiteaSettingsReader.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using NuKeeper.Abstractions;
+using NuKeeper.Abstractions.CollaborationPlatform;
+using NuKeeper.Abstractions.Configuration;
+using NuKeeper.Abstractions.Formats;
+
+namespace NuKeeper.Gitea
+{
+    public class GiteaSettingsReader : ISettingsReader
+    {
+        private readonly IEnvironmentVariablesProvider _environmentVariablesProvider;
+        private const string GiteaTokenEnvironmentVariableName = "NuKeeper_gitea_token";
+        private const string UrlPattern = "https://yourgiteaserver/{owner}/{repo}.git";
+        private const string ApiBaseAdress = "/api/v1/";
+
+        public GiteaSettingsReader(IEnvironmentVariablesProvider environmentVariablesProvider)
+        {
+            _environmentVariablesProvider = environmentVariablesProvider;
+        }
+
+        public Platform Platform => Platform.Gitea;
+
+        public bool CanRead(Uri repositoryUri)
+        {
+            if (repositoryUri == null)
+                return false;
+
+            try
+            {
+                var baseaddress = repositoryUri.OriginalString.Replace(repositoryUri.LocalPath, "");
+
+                // There is no real identifier for gitea repos so try to get the gitea swagger json
+                var client = new HttpClient()
+                {
+                    BaseAddress = new Uri(baseaddress)
+                };
+
+                client.DefaultRequestHeaders.Accept.Add(
+                    new MediaTypeWithQualityHeaderValue("application/json"));
+
+                HttpResponseMessage response = client.GetAsync($"swagger.v1.json").Result;
```

Comment:
Please no `.Result`. This will block the code untill it's complete

Reply:
issue here was CanRead() is no Task<bool>. So I had to go the sync way
Any recommandations how to solve this ?

Reply:
So the only way to know if it's a gitea server is to call the swagger api? There is no convention whatsoever that must be used? If not, i think we should have an extra parameter which specifies that it's the desired platform: `--platform gitea`

Reply:
I think we discuss this approach earlier on gitter or somewhere else.. I don't want to make this a blocker, would be nice if we could add it but it doesn't have to be in this PR. I could be a new upforgrabs feature.

Reply:
`CanRead` - does only provide the repo-uri. Since gitea is most of the time self hosted, the repo uri looks like this
https://localhost:12345/{username}/{projectname}.git

So trying to get the json from https://yourgitea/swagger.v1.json (maybe even trying to validate this json, since there are gitea-version, etc stored) was somehow the safest way.

I think when nukeeper, wants to support self hostet github or vsts services, this issue will appear again sooner or later.

a desired platform-parameter would solve this issue however

Reply:
We have the ` --platform` option already. See https://github.com/NuKeeperDotNet/NuKeeper/wiki/Configuration 

As the docs say:
> This is typicaly infered from the api url structure, but since this does not always work, it can be specified here if neccessary.

Reply:
@AnthonySteele so when using `--platform`, the [CanRead(uri) ](url) will be skipped